File: ProjectEuler/Problems/Problem019.cs
using Microsoft.Extensions.Logging;
using System;

namespace ProjectEuler.Problems
{
    public class Problem019 : Problem<int>
    {
        public Problem019(ILogger<Problem019> logger) : base(logger) { }

        protected override int CalculateAnswer()
        {
            var answer = 0;

            Logger.LogDebug("Incremental Results:");
            for (var year = 1901; year <= 2000; ++year)
            {
                var yearTotal = 0;
                for (var month = 1; month <= 12; ++month)
                {
                    var date = new DateTime(year, month, 1);
                    if (date.DayOfWeek == DayOfWeek.Sunday) ++yearTotal;
                }
                Logger.LogDebug("{year}: {yearTotal}", year, yearTotal);
                answer += yearTotal;
            }

            return answer;
        }
    }
}

File: ProjectEulerPython/problems/problem_019.py
from datetime import date
from problems.problem import Problem

class Problem019(Problem):

  def calculate_answer(self) -> int:
    answer = 0

    self.print_detail("Incremental results:")
    for year in range(1901, 2001):
      year_total = 0
      for month in range(1, 13):
        value = date(year, month, 1)
        if value.weekday() == 6:
          year_total += 1
      self.print_detail(f"{year}: {year_total}")
      answer += year_total

    return answer